node/impl/full.go

```diff
@@ -37,4 +40,40 @@ func (n *FullNodeAPI) CreateBackup(ctx context.Context, fpath string) error {
 	return backup(n.DS, fpath)
 }
 
+func (n *FullNodeAPI) NodeStatus(ctx context.Context) (status api.NodeStatus, err error) {
+	curTs, err := n.ChainHead(ctx)
+	if err != nil {
+		return status, err
+	}
+
+	status.SyncStatus.Epoch = uint64(curTs.Height())
+	timestamp := time.Unix(int64(curTs.MinTimestamp()), 0)
+	delta := time.Since(timestamp).Seconds()
+	status.SyncStatus.Behind = uint64(delta / 30)
+
+	pis, err := n.NetPeers(ctx)
+	if err != nil {
+		return status, err
+	}
+
+	scores, err := n.NetPubsubScores(ctx)
+	if err != nil {
+		return status, err
+	}
+
+	scoreMap := make(map[peer.ID]float64)
+	for _, score := range scores {
+		scoreMap[score.ID] = score.Score.Score
+	}
+
+	for _, pi := range pis {
+		score := scoreMap[pi.ID]
+		if score > -1000 {
+			status.PeerStatus.PeersToPublish++
```

Comment:
Can we break down peers to publish by topic? The message management CLI cares about mpool, not about blocks (I think?).

Reply:
I guess we could list peers by topic and only use those; but in practice all of them subscribe to the same topics as us.

Reply:
We can also break it into two, peers to publish blocks and peers to publish messages.

Reply:
crap now I have to fiddle with  DI :)

Reply:
ah, stmgr seems to be able to give it to me; hooray!

Reply:
done; there are now two separate counts -- one for blocks and once for msgs, and I use the peers listed in each topic by pubsub itself.